src/WebJobs.Script.WebHost/Configuration/ScriptApplicationHostOptionsSetup.cs

```diff
@@ -61,6 +69,69 @@ public void Configure(string name, ScriptApplicationHostOptions options)
                 options.IsSelfHost = options.IsSelfHost;
                 options.IsStandbyConfiguration = true;
             }
+
+            options.AreZipDeploymentAppSettingsValid = ValidateZipDeploymentAppSettings();
+            options.ScmRunFromPackageBlobExists = BlobExists();
+            options.IsZipDeployment = IsZipDeployment(options);
+            options.IsFileSystemReadOnly = options.IsZipDeployment;
+        }
+
+        private bool BlobExists()
+        {
+            var blobExists = _cloudBlockBlobHelperService.BlobExists(_environment.GetEnvironmentVariable(ScmRunFromPackage), EnvironmentSettingNames.ScmRunFromPackage, _logger).GetAwaiter().GetResult();
+            _logger.LogInformation($"Checked if ${EnvironmentSettingNames.ScmRunFromPackage} points to an existing blob: ${blobExists}");
+            return blobExists;
+        }
+
+        private bool IsZipDeployment(ScriptApplicationHostOptions options)
+        {
+            // If the app is using the new app settings for zip deployment, we don't need to check further, it must be a zip deployment.
+            bool usesNewZipDeployAppSettings = IsValidZipSetting(_environment.GetEnvironmentVariable(AzureWebsiteZipDeployment)) ||
+                IsValidZipSetting(_environment.GetEnvironmentVariable(AzureWebsiteAltZipDeployment)) ||
+                IsValidZipSetting(_environment.GetEnvironmentVariable(AzureWebsiteRunFromPackage));
+
+            if (usesNewZipDeployAppSettings)
+            {
+                return usesNewZipDeployAppSettings;
+            }
+
+            // Check if the old app setting for zip deployment is set, SCM_RUN_FROM_PACKAGE.
+            // If not, this is not a zip deployment. If yes, we still need to check if the blob exists,
+            // since this app setting is always created for Linux Consumption, even when it is not used.
+            // In portal deployment, SCM_RUN_FROM_PACKAGE will be set, but we will be using Azure Files instead.
+            if (!IsValidZipUrl(_environment.GetEnvironmentVariable(ScmRunFromPackage)))
+            {
+                return false;
+            }
+
+            // If SCM_RUN_FROM_PACKAGE is a valid app setting and Azure Files are not being used, it must be a zip deployment.
+            if (string.IsNullOrEmpty(_environment.GetEnvironmentVariable(AzureFilesConnectionString)) &&
+                string.IsNullOrEmpty(_environment.GetEnvironmentVariable(AzureFilesContentShare)))
+            {
+                return true;
+            }
+
+            // SCM_RUN_FROM_PACKAGE is set, as well as Azure Files app settings, so we need to check if we are actually using the zip blob.
+            return options.ScmRunFromPackageBlobExists;
+        }
+
+        private bool ValidateZipDeploymentAppSettings()
```

Comment:
ValidateZipDeploymentAppSettings can be moved to Environment since it is just checking the settings are not empty?

Reply:
Good point, will do that.